File: shop_android_kotlin/app/src/main/java/cn/lblbc/shop/module/login/LoginManager.kt
package cn.lblbc.shop.module.login

import cn.lblbc.shop.utils.Constants
import cn.lblbc.shop.utils.SpUtil

object LoginManager {
    fun isLoggedIn() = !SpUtil.get(Constants.SP_KEY_TOKEN, "").isNullOrEmpty()
    fun quitLogin() {
        SpUtil.remove(Constants.SP_KEY_TOKEN)
    }
}

File: shop_flutter/lib/pages/login/login_manager.dart
import 'package:shared_preferences/shared_preferences.dart';
import 'package:shop_flutter/lblbc_constants.dart';

class LoginManager {
  static Future<bool> isLoggedIn() async {
    SharedPreferences prefs = await SharedPreferences.getInstance();
    int? userId = prefs.getInt(Constants.spKeyUserId);
    var result = false;
    if (userId != null) {
      result = true;
    }
    return result;
  }

  static logout() async {
    SharedPreferences prefs = await SharedPreferences.getInstance();
    prefs.remove(Constants.spKeyUserId);
    prefs.remove(Constants.spKeyToken);
  }
}
